### PR TITLE
Fixed this function to measure angles from semi-major axis correctly

### PR DESCRIPTION
**Pull request by drbitboy**
see 
https://github.com/moiseevigor/elliptic/commit/49bc0fe5aa1a3e7f2f0168c63f5855d008950974

### CODE REVIEW

**Comment by drbitboy** [comment]
didn't realize what close did.

File: arclength_ellipse.m [hunk 1]
```diff
@@ -73,15 +73,50 @@
...
+%%%        arclength_ellipse(1, 2., pi/2 - pi*.002, pi/2 - pi*.001) ~ -pi*.001
+%%%        etc.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%% Default solution for a==b (circles)
 arclength = a.*(theta1-theta0);
+
+%%% Ellipses (a<b or a>b)
 if(a<b)
-    [F1, E1] = elliptic12( theta1-theta0, 1 - (a./b).^2 );
```
Comment (drbitboy) [inline arclength_ellipse.m]: theta1-theta0 is an error.

File: arclength_ellipse.m [hunk 2]
```diff
@@ -73,15 +73,50 @@
...
+    %%% Theta measured from a axis = semi-MINOR axis
+    %%% Use standard formulation for E(phi,m)
+    [F1, E1] = elliptic12( theta1, 1 - (a./b).^2 );
     [F0, E0] = elliptic12( theta0, 1 - (a./b).^2 );
     arclength = b.*(E1 - E0);
 elseif(a>b)   
-    [F1, E1] = elliptic12( theta1-theta0, 1 - (b./a).^2 );
```
Comment (drbitboy) [inline arclength_ellipse.m]: theta1-theta0 is an error here; it should be only theta1

**Comment by moiseevigor** [comment]
Thank you Brian for this precious corrections! I'm sure you're correct and beg your pardon for my loose understanding of celestial mechanics, but could you please to point me out to the book/article with the comprehensive studying of the properties on the arc-length? Even tables of calculated arc-length for different values of parameters would nice to check.

I'd like to elaborate on comments in you pull request, give you the credit and create the testing suite for this procedure since it looks like been used quite extensively.

Thanks again!

**Comment by drbitboy** [comment]
https://en.wikipedia.org/wiki/Orbital_elements

There are many others, but that is a decent summary that groups the six elements into their separate effect:  size and shape of the orbit ellipse; orientation of the orbital plane; position of the orbiting body at an epoch on the ellipse.

Here is a page that has more detailed definitions of the various elements:

  http://www.amsat.org/amsat/keps/kepmodel.html

I haven't read it through, but from what I did read seems reasonable.

On the relevance of the major axis of the ellipse:

1) The Argument of the Perihelion (or of Perigee or of Periapse; ArgP or W0 or Wnaught)  this is an angle expressing the orientation within the orbital plane of the ray from the central body to the orbiting body at the time of closest approach.  Since the central body is on the major axis, this ray is along that same axis.

2) There are two forms of anomailies (angles), true and eccentric, to define the position of the body in the orbital plane, but all are measured relative to the vector at periapse i.e. from the sem-major axis.  I am pretty sure that the eccentric anomaly is the same angle as u in elliptic12.m, and as theta0 and theta1 in arclength_ellipse.m.

3) Sometimes the size of the orbit (ellipse) is defined via the length of the semi-major axis (a), and sometimes via the peri-focal distance (Rp), which is along the major axis.  Since the eccentricity is also supplied, and Rp = a * (1-e), it doesn't matter which formulation is used, but the point is that both refer to a distance along the major axis.

3.1) Sometime the size of the ellipse is not specified directly, but the period (time per 360 degrees) or mean daily motion (degrees per time) and the GM parameter (gravitational constant times mass) of the central body is provided.  Again, the Keplerian formula means a and Rp can both be derived from that information.

And here is my by-hand analysis of the derivation of arc length and how to deal with both b>a and a>b:

img_4074

which is similar to this:

  https://www.youtube.com/watch?v=4KdZm4TnUvA

**Pull request merged**